hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -550,22 +551,63 @@ protected InternalScanner createScanner(HStore store, ScanInfo scanInfo,
         dropDeletesFromRow, dropDeletesToRow);
   }
 
-  public List<Path> getCompactionTargets() {
-    T writer = this.writer;
-    if (writer == null) {
-      return Collections.emptyList();
+  /**
+   * Return the progress for a given compaction request.
+   * @param request the compaction request
+   */
+  public CompactionProgress getProgress(CompactionRequestImpl request) {
+    return progressMap.get(request);
+  }
+
+  /**
+   * Return the aggregate progress for all currently active compactions.
+   */
+  public CompactionProgress getProgress() {
+    synchronized (progressMap) {
+      long totalCompactingKVs = 0;
+      long currentCompactedKVs = 0;
+      long totalCompactedSize = 0;
+      for (CompactionProgress progress: progressMap.values()) {
+        totalCompactingKVs += progress.totalCompactingKVs;
+        currentCompactedKVs += progress.currentCompactedKVs;
+        totalCompactedSize += progress.totalCompactedSize;
+      }
+      CompactionProgress result = new CompactionProgress(totalCompactingKVs);
+      result.currentCompactedKVs = currentCompactedKVs;
+      result.totalCompactedSize = totalCompactedSize;
+      return result;
     }
-    if (writer instanceof StoreFileWriter) {
-      return Arrays.asList(((StoreFileWriter) writer).getPath());
+  }
+
+  public boolean isCompacting() {
+    return !progressMap.isEmpty();
+  }
+
+  /**
+   * Return the list of target files for all currently active compactions.
+   */
+  public List<Path> getCompactionTargets() {
+    // Build a list of all the compaction targets for all active writers
+    List<Path> targets = new ArrayList<>();
+    synchronized (writerMap) {
+      for (T writer: writerMap.values()) {
+        if (writer instanceof StoreFileWriter) {
+          targets.add(((StoreFileWriter) writer).getPath());
+        } else {
+          ((AbstractMultiFileWriter) writer).writers().stream()
```

Comment:
Suggest: make this an `else if (writer instance of AbstractMultiFileWriter)` to be defensive and have the `else` branch to fail loudly if something goes wrong.